src/app/services/batch-manager.ts

```diff
@@ -1,21 +1,52 @@
-import { Manager } from 'web3-core-requestmanager';
-import { JsonRPCRequest, Provider } from 'web3/providers';
-import { errors } from 'web3-core-helpers';
+import { AbstractWeb3Module } from 'web3-core';
+import { AbstractMethod } from 'web3-core-method/types';
+import { isObject } from 'rxjs/internal-compatibility';
 
 export interface BatchRequest {
-    readonly request: JsonRPCRequest | any;
+    readonly request: AbstractMethod;
     readonly defaultValue?: any;
 }
 
+export class Validator {
+    static validate(response, payload?: object) {
+        if (isObject(response)) {
+            if (response.error) {
+                if (response.error instanceof Error) {
+                    return new Error(`Node error: ${response.error.message}`);
+                }
+
+                return new Error(
+                    `Node error: ${JSON.stringify(response.error)}`
+                );
+            }
+
+            if (payload && response.id !== payload['id']) {
+                return new Error(
+                    `Validation error: Invalid JSON-RPC response ID (request: ${
+                        payload['id']
+                    } / response: ${response.id})`
+                );
+            }
+
+            if (response.result === undefined) {
+                return new Error('Validation error: Undefined JSON-RPC result');
+            }
+
+            return true;
+        }
+
+        return new Error('Validation error: Response should be of type Object');
+    }
+}
+
 export class BatchManager {
+    constructor(moduleInstance: AbstractWeb3Module) {
```

Comment:
Just a suggestion for next time, not needed now, but TS allows you to receive, declare visibility and set it on constructor parameters through `constructor(private moduleInstance: AbstractWeb3Module)` in place of the trivial implementation

Reply:
I will fix it no worries, I just missed it.